src/api/code-challenge/content-types/code-challenge/lifecycles.js

```diff
@@ -0,0 +1,81 @@
+module.exports = {
+  beforeCreate(event) {
+    console.log("**** beforeCreate hook is activated! ****");
+    const [eventTests, eventMetaTests] = [
+      event.params.data.tests,
+      event.params.data.MetaTest,
+    ];
+    runMetaTests(eventTests, eventMetaTests);
+  },
+
+  beforeUpdate(event) {
+    console.log("**** beforeUpdate hook is activated! ****");
+    const [eventTests, eventMetaTests] = [
+      event.params.data.tests,
+      event.params.data.MetaTest,
+    ];
+    runMetaTests(eventTests, eventMetaTests);
+  },
+};
+
+// runMetaTestsNeed more specific name later, maybe use "validateInternalTests"
+async function runMetaTests(eventTests, eventMetaTests) {
+  const [tests, metaTests] = await Promise.all([
+    getInternalTests(eventTests),
+    getMetaTests(eventMetaTests),
+  ]);
+
+  tests.map((test) =>
+    console.log("final test.internalTest: ", test.internalTest)
+  );
+
+  metaTests.map((metaTest) =>
+    console.log("final metaTest caseCode: ", metaTest.caseCode)
+  );
+
+  return false;
+}
+
+async function getInternalTests(eventTests) {
+  const internalTests = await strapi.db
+    .query("challenge.code-challenge-test")
+    .findMany({
+      select: ["id", "internalTest"],
+      where: {
+        id: getCurrentTestIDs(eventTests),
+      },
+    });
+
+  // console.log("internalTests: ", internalTests);
+  return internalTests;
+}
+
+function getCurrentTestIDs(eventTests) {
+  let currentTestIDs = [];
+  eventTests.map((test) => {
+    currentTestIDs.push(test.id);
+  });
```

Comment:
this defeats the point of map. You need to return inside of the map function. Take a look at map on MDN

Reply:
function getCurrentTestIDs(eventTests) {
  return eventTests.map(({id}) => id);
}

Reply:
The eventTests array is an array of objects like 
`[{id: 10}, {id: 11}]` 
I am trying to create a new array of ids like this:
`[10, 11]`

Reply:
Thanks for the heads up. I didn't realize .map() could be so flexible!